src/dev/server.ts

```diff
@@ -158,14 +158,16 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
   // Main worker proxy
   const proxy = createProxy();
   proxy.proxy.on("proxyReq", (proxyReq, req) => {
-    // TODO: Avoid overwriting headers if already set
-    if (req.socket.remoteAddress) {
+    if (req.socket.remoteAddress && !proxyReq.getHeaders()["x-forwarded-for"]) {
       proxyReq.setHeader("X-Forwarded-For", req.socket.remoteAddress);
     }
-    if (req.socket.remotePort) {
+    if (req.socket.remotePort && !proxyReq.getHeaders()["x-forwarded-port"]) {
```

Comment:
Perf: We don't need to call getHeaders twice. Instead, we can keep it in a variable before L160

Reply:
I'll fix

Reply:
I have committed a fix. I don't think we can do it before L160, since we don't have `proxyReq` available

Reply:
That's a right place